Assets/HoloToolkit/UX/Scripts/Dialog/DialogShell.cs

```diff
@@ -0,0 +1,249 @@
+                                    button.gameObject.SetActive(buttonTypes.Count == 1);
+
+                                    if (buttonTypes.Count == 1)
+                                    {
+                                        oneButtonSet[0] = button.gameObject;
+                                        button.SetTitle(buttonTypes[0].ToString());
+                                        button.ButtonTypeEnum = buttonTypes[0];
+                                    }
+                                }
+                                else if (button.name.Contains("ButtonTwo"))
+                                {
+                                    button.gameObject.SetActive(buttonTypes.Count > 1);
+
+                                    if (buttonTypes.Count > 1)
+                                    {
+                                        if (button.name.Contains("A"))
+                                        {
+                                            twoButtonSet[0] = button.gameObject;
+                                            button.SetTitle(buttonTypes[0].ToString());
+                                            button.ButtonTypeEnum = buttonTypes[0];
+                                        }
+                                        else if (button.name.Contains("B"))
+                                        {
+                                            twoButtonSet[1] = button.gameObject;
+                                            button.SetTitle(buttonTypes[1].ToString());
+                                            button.ButtonTypeEnum = buttonTypes[1];
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        protected override void SetTitleAndMessage()
+        {
+            foreach (Transform child in transform)
+            {
+                if (child != null && child.name == "TitleText")
+                {
+                    titleText = child.GetComponent<TextMesh>();
+                }
+                else if( child != null && child.name == "TitleMessage")
+                {
+                    messageText = child.GetComponent<TextMesh>();
+                }
+            }
+
+            if (titleText != null)
+            {
+                titleText.text = Result.Title;
+            }
+
+            if (messageText != null)
+            {
+                messageText.text = WordWrap(Result.Message, MaxCharsPerLine);
+            }
+        }
+
+        public static string WordWrap(string text, int maxCharsPerLine)
+        {
+            int pos = 0;
+            int next = 0;
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (maxCharsPerLine < 1)
+                return text;
+            
+            for (pos = 0; pos < text.Length; pos = next) {
```

Comment:
Please move the { from the end of this line to the next (like other loops and conditionals in this file)